### PR TITLE
PRO-2121 PRO-2353 support for overriding Vue components at project level

### LINKED ISSUE #3465: Replacing a UI component in project code throws an error
## To Reproduce

In project code, place a Vue file at the same relative path as a core UI component, copy the original code and make a small change to identify the override.

## Expected behavior
The project-level file should be used instead of the core component.

## Describe the bug
Currently this is causing a conflict and throwing an error:
```
Uncaught Error: Module parse failed: Identifier 'TheAposAdminBarUser' has already been declared (278:21)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|               
| 
>               import TheAposAdminBarUser from "/Users/abea/Sites/testbed/modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarUser.vue";
|               
| 
    js http://localhost:3000/apos-frontend/default/apos-module-bundle.js:279

```

—

Created via [Raycast](https://www.raycast.com?ref=signatureGithub)

### PR DESCRIPTION
**Pull request by boutell**
Resolves #3465

### CODE REVIEW

**Comment by boutell** [comment]
See the `custom-logo-vue-component` of our testbed project for a demo in which the apostrophe logo in the admin bar (not the one seen when logged out) is overridden.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## UNRELEASED
+
+### Adds
+
+* Developers can now override any Vue component of the ApostropheCMS admin UI by providing a component of the same name in the `ui/apos/components` folder of their own module. Apostrophe will use the version provided by the last module initialized, so a module that uses `improve` will beat a base class, a project level module will beat a module that uses `improve`, and so on. Bear in mind however that there can only be one implementation of a component with a given name that is actually active in your project. If you need different behavior for different piece types, take advantage of the `components` option offered by many modules to configure a specific Vue component for a specific role, for instance: `{ browserOptions: { components: 'MyCustomEditorModal' } }`
```
Comment (abea) [inline CHANGELOG.md:7]: Let's try to keep it to a changelog. We should definitely have a docs update with this as well.

**Review by abea — CHANGES_REQUESTED** [review]
It works beautifully. We will need separate documentation and should keep the changelog to the change.

**Comment by boutell** [comment]
Dialed it way back but left a little hint that there are other ways.

**Review by abea — APPROVED** [review]

**Pull request merged**